modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/TransactionProxyImpl.java

```diff
@@ -421,4 +421,63 @@ private void saveFuture(IgniteInternalFuture<IgniteInternalTx> fut) {
     @Override public String toString() {
         return S.toString(TransactionProxyImpl.class, this);
     }
+
+    /** {@inheritDoc} */
+    @Override public void savepoint(String name) throws IllegalArgumentException {
+        savepoint(name, false);
+    }
+
+    /** {@inheritDoc} */
+    @Override public void savepoint(String name, boolean overwrite) throws IllegalArgumentException {
+        if (implicit())
+            throw new IllegalStateException("Savepoints must be used inside explicit transactions.");
```

Comment:
"Savepoints can be used only inside explicit transactions." sound better

Reply:
may be change parameter "name" to "savepointId" .

Reply:
"SavepointId" looks like it is number, not meaningful string. "savepointName" is redundant to me - method operates on savepoints, so name it takes is name of savepoint.